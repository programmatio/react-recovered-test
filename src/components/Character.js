import React from "react";
import PropTypes from "prop-types";

const Character = ({ character }) => {
  return (
    <dl>
      <dt>Name</dt>
      <dd>{character.name}</dd>
      <dt>Heigh</dt>
      <dd>{character.height}</dd>
      <dt>Hair Color</dt>
      <dd>{character.hair_color}</dd>
      <dt>Eye Color</dt>
      <dd>{character.eye_color}</dd>
    </dl>
  );
};
Character.propTypes = {
  character: PropTypes.object.isRequired
};

export default Character;
