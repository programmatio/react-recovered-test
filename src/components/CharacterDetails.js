import React from "react";
import PropTypes from "prop-types";

import Character from "./Character";
const CharacterDetails = ({ characters }) => {
  return (
    <div>
      <h2>Characters:</h2>
      {!!characters.length &&
        characters.map((character, index) => {
          return <Character key={index} character={characters[index]} />;
        })}
      {!characters.length && <p>Nothing to display</p>}
    </div>
  );
};

CharacterDetails.props = {
  characters: PropTypes.array.isRequired
};

export default CharacterDetails;
