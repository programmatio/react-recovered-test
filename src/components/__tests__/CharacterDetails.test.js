import React from "react";
import { shallow } from "enzyme";
import CharacterDetails from "../CharacterDetails";
import Character from "../Character";

describe("<CharacterDetails /> without Characters", () => {
  const props = {
    characters: []
  };
  const wrapper = shallow(<CharacterDetails {...props} />);
  it("should match the snapshot", function() {
    expect(wrapper).toMatchSnapshot();
  });
  it("should not have any definition lists without any characters in the array", function() {
    expect(wrapper.find(Character).length).toBe(0);
  });
  it('should have text paragraph with "Nothing to display" message', function() {
    expect(wrapper.find("p").text()).toEqual("Nothing to display");
  });
});

describe("<CharacterDetails /> with Characters", () => {
  const props = {
    characters: [
      {
        name: "test",
        height: "test",
        hair_color: "test",
        eye_color: "test"
      },
      {
        name: "test",
        height: "test",
        hair_color: "test",
        eye_color: "test"
      }
    ]
  };

  const wrapperWithCharacters = shallow(<CharacterDetails {...props} />);

  it("should show characters", function() {
    expect(wrapperWithCharacters.find(Character).length).toBe(2);
  });
  it('should not have text paragraph with "Nothing to display" message', function() {
    expect(wrapperWithCharacters.find("p").length).toBe(0);
  });
});
