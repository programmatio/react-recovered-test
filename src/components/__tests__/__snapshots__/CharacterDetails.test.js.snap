// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CharacterDetails /> without Characters should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CharacterDetails
    characters={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Characters:
</h2>,
        false,
        <p>
          Nothing to display
</p>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Characters:",
        },
        "ref": null,
        "rendered": "Characters:",
        "type": "h2",
      },
      false,
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Nothing to display",
        },
        "ref": null,
        "rendered": "Nothing to display",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Characters:
</h2>,
          false,
          <p>
            Nothing to display
</p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Characters:",
          },
          "ref": null,
          "rendered": "Characters:",
          "type": "h2",
        },
        false,
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Nothing to display",
          },
          "ref": null,
          "rendered": "Nothing to display",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
